### DISCUSSION TITLE
Allow lower php versions than 7.3

### DISCUSSION THREAD

**Discussion by c33s** [composer-unused/composer-unused#141] [General]
at least 7.1 would be very nice for an external tool. 

for my libary i test against multiple php versions and requiring 7.3 for this tools breaks every pipeline run against a lower php version.

**Comment by icanhazstring** [comment]
Hi @c33s, php 7.2 support was dropped in Nov 2020. So my minimum requirement currently is 7.3.

I currently don't plan on providing or testing for 7.1.

**Reply by c33s** [reply]
from my perspective a ci tool, like this one, should really be able to run on outdated php versions. i even have to still support php 5 (!) projects. so my ci chain works from 5.6 to 7.4 with a planned extension to 8.0 soon.
if ci tools require nearly latest php versions then it is really a problem to keep libs compatible.

would be nice to rethink you decision

**Reply by icanhazstring** [reply]
It's not like I can't do it. But rather a thing of priority. While I develop this tool in my spare time I need to consider what needs to be done to get it going forward.

Sure having support for older version might be useful in some cases. On the other hand I won't receive a dime doing it. So one way would be to sponsor this kind of feature so I can even compensate the time spent adding it.

**Reply by SvenRtbg** [reply]
This tool is analysing the code statically, it does not use PHP to run your code. Which would allow to create a PHP runtime island with the latest-and-greatest PHP that is only used to execute this tool. The answer will very likely be the same regardless of which PHP you use.
